client/state/comments/reducer.js

```diff
@@ -509,6 +530,26 @@ export const counts = ( state = {}, action ) => {
 			);
 			return Object.assign( {}, state, { [ siteId ]: newTotalSiteCounts } );
 		}
+		case COMMENTS_EMPTY_RECEIVE: {
+			const { siteId, status, commentIds } = action;
+
+			if ( ! siteId || ! state[ siteId ] || ! status ) {
+				return state;
+			}
+			const { site: siteCounts } = state[ siteId ];
+
+			const emptiedCommentsCount = commentIds?.length || 0;
+			const newSiteCounts = updateCount( siteCounts, status, -emptiedCommentsCount );
+
+			// Post counts can't be updated here because we don't know the post ID.
```

Comment:
My findings on this are:

* Comment counts are only used by Comment Management (`getSiteCommentCounts` selector)
* Comment Management requests fresh counts via a query component on each load of the comment list (via `QuerySiteCommentCounts`)
* Spam and trash comments are not included in the comment count that is displayed on the post list (e.g. /posts/example.com)

...so there seem to be no adverse effects of having stale spam/trash post counts in state, even if that doesn't feel particularly nice.

Reply:
We could trigger some re-fetching of comment counts on successful spam cleanup, can't we?

Reply:
Sounds like a plan. What's the cleanest way to do that? via an additional prop to the query component?

Reply:
Good idea. I had a play with firing a `COMMENT_COUNTS_REQUEST` from the data layer but I think that just gives us the same site comment counts we already have. We also re-fetch comments (filtered for the current comment list) when the emptying completes via `refreshCommentListQuery`:

https://github.com/Automattic/wp-calypso/blob/260f745ebb798039c11c56950fe501638c48d0c2/client/state/data-layer/wpcom/comments/index.js#L239

I think that should be enough. 🤞